### PR TITLE
add package MJPG-streamer

### PR DESCRIPTION
**Pull request by bobvanderlinden**
MJPG-streamer takes JPGs from Linux-UVC compatible webcams, filesystem or other input plugins and streams them as M-JPEG via HTTP to webbrowsers, VLC and other software. It is the successor of uvc-streamer, a Linux-UVC streaming application with Pan/Tilt.

MJPG-streamer does not have a configuration step (just a bare makefile), so `gspcav` support needed to be disabled by altering the `Makefile`.
Also MJPG-streamer is build using dynamically linked plugins. On ArchLinux these are placed in `/usr/lib/`, so the linker can always find these, here I've used the `-rpath` option and enabled `dontPatchELF`. Let me know if there is a better way to do this.

### CODE REVIEW

**Comment by lethalman** [comment]
I think disabling patchelf is wrong. You could patchelf in `postFixup` by retrieving the rpath, adding lib/plugins, and setting back the rpath.

**Comment by bobvanderlinden** [comment]
Thanks for the suggestion. It now uses `postFixup`.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -9406,6 +9406,8 @@ let
     inherit (vamp) vampSDK;
   };
 
+  mjpg_streamer = callPackage ../applications/video/mjpg-streamer { };
```
Comment (lethalman) [inline pkgs/top-level/all-packages.nix]: Last thing, what about `mjpg-streamer` here as attribute name?

Comment (bobvanderlinden) [inline pkgs/top-level/all-packages.nix]: I saw some of the gnome packages used "-" in the filename, but "_" as the
attribute name, so I went with that. `mjpg-streamer` is indeed more
consistent in this case, so I've changed it.

**Comment by lethalman** [comment]
"_" is usually old naming.

**Comment by lethalman** [comment]
Travis passed, thanks.

**Comment by bobvanderlinden** [comment]
Ah, good to know. Thanks for merging!

**Pull request merged**